webapp/channels/src/components/login/login.tsx

```diff
@@ -129,22 +131,31 @@ const Login = ({onCustomizeHeader}: LoginProps) => {
     const enableSignInWithEmail = EnableSignInWithEmail === 'true';
     const enableSignInWithUsername = EnableSignInWithUsername === 'true';
     const enableSignUpWithEmail = EnableSignUpWithEmail === 'true';
-    const enableSignUpWithGitLab = EnableSignUpWithGitLab === 'true';
-    const enableSignUpWithGoogle = EnableSignUpWithGoogle === 'true';
-    const enableSignUpWithOffice365 = EnableSignUpWithOffice365 === 'true';
-    const enableSignUpWithOpenId = EnableSignUpWithOpenId === 'true';
+    const enableSignUpWithGitLab = true;
+    const enableSignUpWithGoogle = true;
+    const enableSignUpWithOffice365 = true;
+    const enableSignUpWithOpenId = true;
     const isLicensed = IsLicensed === 'true';
     const ldapEnabled = isLicensed && enableLdap;
     const enableSignUpWithSaml = isLicensed && enableSaml;
     const siteName = SiteName ?? '';
 
     const enableBaseLogin = enableSignInWithEmail || enableSignInWithUsername || ldapEnabled;
-    const enableExternalSignup = enableSignUpWithGitLab || enableSignUpWithOffice365 || enableSignUpWithGoogle || enableSignUpWithOpenId || enableSignUpWithSaml;
+    const enableExternalSignup = true;
     const showSignup = enableOpenServer && (enableExternalSignup || enableSignUpWithEmail || enableLdap);
 
     const query = new URLSearchParams(search);
     const redirectTo = query.get('redirect_to');
 
+    const handleExternalBtnFocus = () => {
```

Comment:
you can simplify this to be only one function and then use it this way:

```
const handleExternalBtnFocusBlur = (isFocused: boolean) => {
    setIsExternalBtnFocused(isFocused)
}

```

And then use it like this:

```
<ExternalLoginButton
    ...
    onFocus={() => handleExternalBtnFocusBlur(true)}
    onBlur={() => handleExternalBtnFocusBlur(false)}
    ...
/>
```

also, the function should be memoized using the useCallback function to avoid unnecessary re-renders of the child component.